docs/references/t-ecdsa-how-it-works.mdx

```diff
@@ -1,73 +1,110 @@
+The ECDSA API comprises two methods: `ecdsa_public_key` for retrieving threshold ECDSA public keys, and `sign_with_ecdsa` for requesting threshold ECDSA signatures to be computed from the subnet holding the secret-shared private threshold ECDSA key.
+The Schnorr API comprises two methods: `schnorr_public_key` for retrieving threshold Schnorr public keys, and `sign_with_schnorr` for requesting threshold schnorr signatures to be computed from the subnet holding the secret-shared private threshold Schnorr key.
 
-Each API call refers to a threshold ECDSA master key by virtue of a 2-part identifier comprising a curve and a key id as outlined above. Derivation paths are used to refer to keys below a canister's root key in the key derivation hierarchy. The key derivation from the master key to the canister root key is implicit in the API.
+Each API call refers to a t-sig master key by virtue of a 2-part identifier comprising a curve or algorithm, and a key id as outlined above. Derivation paths are used to refer to keys below a canister's root key in the key derivation hierarchy. The key derivation from the master key to the canister root key is implicit in the API.
 
--   [`ecdsa_public_key`](/docs/current/references/ic-interface-spec#ic-ecdsa_public_key): this method returns a SEC1-encoded ECDSA public key for the given canister using the given derivation path. If the `canister_id` is unspecified, it will default to the canister id of the caller. The `derivation_path` is a vector of variable length byte strings. The `key_id` is a struct specifying both a curve and a name. The availability of a particular `key_id` depends on implementation.<br/>
-For curve `secp256k1`, the public key is derived using a generalization of BIP32 (see [ia.cr/2021/1330](https://eprint.iacr.org/2021/1330), Appendix D). To derive (non-hardened) BIP-0032-compatible public keys, each byte string (blob) in the `derivation_path` must be a 4-byte big-endian encoding of an unsigned integer less than 2<sup>31</sup>.<br/>
+-   [`ecdsa_public_key`](/docs/current/references/ic-interface-spec#ic-ecdsa_public_key): this method returns a SEC1-encoded ECDSA public key for the given canister using the given derivation path. 
+If the `canister_id` is unspecified, it will default to the canister id of the caller. 
+The `derivation_path` is a vector of variable length byte strings. 
+The `key_id` is a struct specifying both a curve and a name. 
+The availability of a particular `key_id` depends on implementation.<br/>
+For curve `secp256k1`, the public key is derived using a generalization of BIP32 (see [ia.cr/2021/1330](https://eprint.iacr.org/2021/1330), Appendix D). 
+To derive (non-hardened) BIP-0032-compatible public keys, each byte string (blob) in the `derivation_path` must be a 4-byte big-endian encoding of an unsigned integer less than 2<sup>31</sup>.<br/>
```

Comment:
```suggestion
-   [`ecdsa_public_key`](/docs/current/references/ic-interface-spec#ic-ecdsa_public_key): This method returns a SEC1-encoded ECDSA public key for the given canister using the given derivation path. 
- If the `canister_id` is unspecified, it will default to the canister id of the caller. 
- The `derivation_path` is a vector of variable length byte strings. 
- The `key_id` is a struct specifying both a curve and a name. The availability of a particular `key_id` depends on implementation.

For curve `secp256k1`, the public key is derived using a generalization of BIP32 (see [ia.cr/2021/1330](https://eprint.iacr.org/2021/1330), Appendix D). To derive (non-hardened) BIP-0032-compatible public keys, each byte string (blob) in the `derivation_path` must be a 4-byte big-endian encoding of an unsigned integer less than 2<sup>31</sup>.
```